### PR TITLE
remove ability to pass in custom mustgather images

### PR DESCRIPTION
**Pull request by AlexVulaj**
Follow up to OSD-21667 - we no longer need the ability to provide custom mustgather images.

### CODE REVIEW

**Comment by AlexVulaj** [comment]
/hold

I haven't been able to test this yet with the proxy outage for backplane access, but the code changes should still be reviewable at this time.

File: build/templates/job.template.yaml [hunk 1]
```diff
@@ -39,12 +38,11 @@ spec:
             echo "Gather timed out."
             exit 0
           fi
-        image: {{ $element }}
-        name: gather-{{ $index }}
+        image: quay.io/openshift/origin-must-gather:latest
```
Comment (mjlshen) [inline build/templates/job.template.yaml:41]: Looking at the available images, https://quay.io/repository/openshift/origin-must-gather?tab=tags&tag=latest Do we want to template this out to use the version of the cluster instead of `:latest`

**Review by mjlshen — COMMENTED** [review]
Looks good overall, added a comment for consideration, not sure if we need it

File: build/templates/job.template.yaml (hunk 1 above)
Comment (AlexVulaj) [reply build/templates/job.template.yaml:41]: This makes sense to me. I'll look into making this happen.

File: controllers/mustgather/mustgather_controller.go [hunk 2]
```diff
@@ -399,6 +385,29 @@ func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.M
...
+		if history.State == "Completed" {
+			version = history.Version
+			break
+		}
+	}
+	if version == "" {
+		return "", goerror.New("unable to determine cluster version from status history")
+	}
+
+	version = version[:strings.LastIndex(version, ".")]
```
Comment (mjlshen) [inline controllers/mustgather/mustgather_controller.go:406]: Can we unit test this function? Small edge case, but I think this doesn't work when the version is something like `4.14.0-rc.0`

You can use https://github.com/openshift/managed-cluster-validating-webhooks/blob/f0862cb7bf1869b00e1ab857ee08fcf6fc7f156c/pkg/webhooks/podimagespec/podimagespec_test.go as an example if you haven't setup a fake kube client before.

File: controllers/mustgather/mustgather_controller.go [hunk 3]
```diff
@@ -399,6 +385,29 @@ func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.M
...
+
+	return version, nil
+}
```
Comment (mjlshen) [inline controllers/mustgather/mustgather_controller.go:409]: Make sure to add RBAC so that the operator is able to get `config.openshift.io/clusterversion` objects https://github.com/openshift/must-gather-operator/blob/master/deploy/02_must-gather-operator.ClusterRole.yaml

File: build/templates/job.template.yaml [hunk 4]
```diff
@@ -18,13 +18,11 @@ spec:
         - effect: NoSchedule
           key: node-role.kubernetes.io/infra
           operator: Exists
-      backoffLimit: 6
```
Comment (mjlshen) [inline build/templates/job.template.yaml:21]: Do we want to remove this (just wondering if it was an intentional change or accidental)

**Review by mjlshen — COMMENTED** [review]
Looks a lot better, thanks for getting the version of the image that is the same as the cluster's version.

* Two requests around how we are getting the clusterversion that I think are important to address.
* I could be convinced around removing the existing `backOffLimit` value and am just asking out of curiosity.

File: build/templates/job.template.yaml (hunk 4 above)
Comment (AlexVulaj) [reply build/templates/job.template.yaml:21]: `backoffLimit` isn't a real field on `.spec.toleration.spec` for a job, plus I was seeing warnings about it when testing so I just removed it.

**Review by mjlshen — APPROVED** [review]
/lgtm

**Review by mjlshen — APPROVED** [review]
/lgtm
